app/src/main/java/ch/epfl/sweng/swenggolf/offer/ShowOfferActivity.java

```diff
@@ -500,9 +488,72 @@ public void closeOffer() {
         Button closeButton = inflated.findViewById(R.id.close_offer_button);
         closeButton.setVisibility(View.GONE);
         getActivity().invalidateOptionsMenu();
+        offerAccessToDiscussion();
         listAnswerAdapter.closeAnswers();
-        //TODO : add listener
     }
 
 
+    private void setClosingListener() {
+        closingListener = new ValueListener<Boolean>() {
+            @Override
+            public void onDataChange(Boolean value) {
+                if (value != null && value) {
+                    hideReactButton();
+                }
+            }
+
+            @Override
+            public void onCancelled(DbError error) {
+                //No notification or errored notifications, as such no modification on display.
+            }
+        };
+        Database.getInstance().listen(Database.OFFERS_PATH + "/" + offer.getUuid(),
+                "isClosed", closingListener, Boolean.class);
+    }
+
+    private void hideReactButton() {
+        Button reactButton = inflated.findViewById(R.id.react_button);
+        if (reactButton != null) {
+            reactButton.setVisibility(View.GONE);
+        }
+        TextView closedMessage = inflated.findViewById(R.id.offer_is_closed);
+        closedMessage.setVisibility(View.VISIBLE);
+    }
+
+    private void offerAccessToDiscussion() {
+        Answers answers = listAnswerAdapter.getAnswers();
+        if (answers.getFavoritePos() != Answers.NO_FAVORITE) {
+            final String chosenUserId = answers.getUserOfPosition(answers.getFavoritePos());
+            if (userIsCreator || chosenUserId.equals(Config.getUser().getUserId())) {
+                Button discussionButton = inflated.findViewById(R.id.open_discussion);
+                discussionButton.setVisibility(View.VISIBLE);
+                discussionButton.setOnClickListener(new View.OnClickListener() {
+                    @Override
+                    public void onClick(View v) {
+                        goToMessagingWithUser(chosenUserId);
+
+
+                    }
+                });
+            }
+        }
+    }
+
+    private void goToMessagingWithUser(String chosenUserId) {
+        String otherUserId = userIsCreator ? chosenUserId : offer.getUserId();
+        Database.getInstance().read(Database.USERS_PATH, otherUserId,
+                new ValueListener<User>() {
+                    @Override
+                    public void onDataChange(User value) {
+                        replaceCentralFragment(
```

Comment:
Have you checked that the back button doesn't mess up the "fragments to skip" count ?

Reply:
I did not think about that, and it's not a change I did this week to be fair. I tried it with a normal situation and it works, but I didn't try to edit the offer, click on an answer, re-edit, and go back, for instance. I will ask if this a problem to @SFulpius and if it is, we can change it later.

Reply:
Okay, let's merge it like this now and we will check this out another day.